### PR TITLE
Autogen jwtIssuer

### PR DESCRIPTION
**Pull request by kfox1111**
Unset a default for jwtIssuer and global.spire.jwtIssuer. When unset, generate it to the default of oidc-discovery.$trustDomain so in many cases the user doesn't need to set it at all.

### CODE REVIEW

**Review by mrsabath — APPROVED** [review]
Tested, it works fine! LGTM!

File: charts/spire/charts/spiffe-oidc-discovery-provider/values.yaml [hunk 1]
```diff
@@ -125,8 +125,8 @@ insecureScheme:
       #   cpu: 100m
       #   memory: 64Mi
 
-## @param jwtIssuer Path to JWT issuer
-jwtIssuer: https://oidc-discovery.example.org
+## @param jwtIssuer Path to JWT issuer. Defaults to oidc-discovery.$trustDomain
```
Comment (faisal-memon) [inline charts/spire/charts/spiffe-oidc-discovery-provider/values.yaml:128]:
```suggestion
## @param jwtIssuer Path to JWT issuer. Defaults to oidc-discovery.$trustDomain if unset
```

File: charts/spire/charts/spire-server/values.yaml [hunk 2]
```diff
@@ -174,8 +174,8 @@ dataStore:
 
 ## @param logLevel The log level, valid values are "debug", "info", "warn", and "error"
 logLevel: info
-## @param jwtIssuer The JWT issuer domain
-jwtIssuer: https://oidc-discovery.example.org
+## @param jwtIssuer The JWT issuer domain. Defaults to oidc-discovery.$trustDomain
```
Comment (faisal-memon) [inline charts/spire/charts/spire-server/values.yaml:177]:
```suggestion
## @param jwtIssuer The JWT issuer domain. Defaults to oidc-discovery.$trustDomain if unset
```

File: charts/spire/values.yaml [hunk 3]
```diff
@@ -13,8 +13,8 @@ global:
     bundleConfigMap: ""
     ## @param global.spire.clusterName The name of the k8s cluster for Spire install
     clusterName: example-cluster
-    ## @param global.spire.jwtIssuer The issuer for Spire JWT tokens
-    jwtIssuer: oidc-discovery.example.org
+    ## @param global.spire.jwtIssuer The issuer for Spire JWT tokens. Defaults to oidc-discovery.$trustDomain
```
Comment (faisal-memon) [inline charts/spire/values.yaml:16]:
```suggestion
    ## @param global.spire.jwtIssuer The issuer for Spire JWT tokens. Defaults to oidc-discovery.$trustDomain if unset
```

File: charts/spire/charts/spiffe-oidc-discovery-provider/values.yaml (hunk 1 above)
Comment (kfox1111) [reply charts/spire/charts/spiffe-oidc-discovery-provider/values.yaml:128]: Looks good. :)

**Review by faisal-memon — APPROVED** [review]

**Pull request merged**